### DISCUSSION TITLE
Missing error handler on socket

### DISCUSSION THREAD

**Discussion by El-Tommy** [socketio/socket.io#4841] [Q&A] [answered]
I have been receiving this alert on my server. What can cause it and how can I fix it?

0|ChatServ | Missing error handler on `socket`.
0|ChatServ | Error: invalid payload
0|ChatServ |     at Decoder.decodeString (/var/chat-server/node_modules/socket.io-parser/build/cjs/index.js:219:23)
0|ChatServ |     at Decoder.add (/var/chat-server/node_modules/socket.io-parser/build/cjs/index.js:123:27)
0|ChatServ |     at Client.ondata (/var/chat-server/node_modules/socket.io/dist/client.js:182:26)
0|ChatServ |     at Socket.emit (node:events:514:28)
0|ChatServ |     at Socket.onPacket (/var/chat-server/node_modules/engine.io/build/socket.js:116:22)
0|ChatServ |     at Polling.emit (node:events:514:28)
0|ChatServ |     at Polling.onPacket (/var/chat-server/node_modules/engine.io/build/transport.js:92:14)
0|ChatServ |     at callback (/var/chat-server/node_modules/engine.io/build/transports/polling.js:164:18)
0|ChatServ |     at Array.forEach (<anonymous>)
0|ChatServ |     at Polling.onData (/var/chat-server/node_modules/engine.io/build/transports/polling.js:170:45)

**Comment by darrachequesne** [comment]
Hi! Which version of `socket.io` are you using? This log should no longer be emitted since version 4.6.2 (https://github.com/socketio/socket.io/commit/15af22fc22bc6030fcead322c106f07640336115).

**Reply by El-Tommy** [reply]
Version: 4.5.2

**Reply by darrachequesne** [reply, marked as answer]
Then you can either:

- upgrade to the latest version of `socket.io` (currently `4.7.2`)
- or add an empty 'error' handler:

```js
socket.on("error", () => {});
```

**Reply by El-Tommy** [reply]
Thank you for your help.

One question, can I use the same socket.io-client from 4.5.2 to 4.7.2 or should I update as well?

In my projects I always started with a version and never updated it, which is why this other very silly question: How do I update the Socket.IO Server version on NPM?

And one more question, when an "Error: invalid payload" occurs, can I see the payload (or even the event) that was received by the server when using the code below?

`socket.on("error", () => {});`

**Reply by darrachequesne** [reply]
> One question, can I use the same socket.io-client from 4.5.2 to 4.7.2 or should I update as well?

You can keep using version `4.5.2` or upgrade to `4.7.2`, as it's only minor changes. Major changes (from `2.x.x` to `4.x.x` for example) would require more caution.

Reference: https://socket.io/docs/v4/changelog/

> How do I update the Socket.IO Server version on NPM?

You can use `npm update socket.io`.

Reference: https://docs.npmjs.com/cli/v6/commands/npm-update

> And one more question, when an "Error: invalid payload" occurs, can I see the payload (or even the event) that was received by the server when using the code below?

No, currently this is not possible.

**Reply by El-Tommy** [reply]
Thank you very much.
